src/controllers/signupController.js

```diff
@@ -0,0 +1,43 @@
+import bcrypt from 'bcrypt';
+import dotenv from 'dotenv';
+import jwt from 'jsonwebtoken';
+import SignupService from '../services/signupService';
+
+dotenv.config();
+/**
+ * @exports
+ * @class UsersController
+ */
+class UsersController {
+  /**
+ * users can create an account
+ * @static
+ * @description POST /api/v1/signUp
+ * @param {object} req request object
+ * @param {object} res response object
+ * @memberof UsersController
+ * @returns {object} data
+ */
+  static async signUp(req, res) {
+    const userExist = await SignupService.checkUserExistByEmail(req.body.email);
+    if (!userExist) {
+      const hashedPassword = bcrypt.hashSync(req.body.password, 10);
+      req.body.password = hashedPassword;
+      const createdUser = await SignupService.addUser(req.body);
+      const {
+        id, firstName, lastName, email, username, role, isVerified
+      } = createdUser;
+      const token = jwt.sign({
```

Comment:
During signup, you shouldn't generate a token for the user. Leave it to the login feature. I gave feedback regarding this during a standup